homeassistant/components/rest/sensor.py

```diff
@@ -229,22 +229,22 @@ def __init__(
         """Initialize the data object."""
         self._request = requests.Request(
             method, resource, headers=headers, auth=auth, data=data
-        ).prepare()
+        )
         self._verify_ssl = verify_ssl
         self._timeout = timeout
         self.data = None
 
     def set_url(self, url):
         """Set url."""
-        self._request.prepare_url(url, None)
+        self._request.url = url
 
     def update(self):
         """Get the latest data from REST service with provided method."""
         _LOGGER.debug("Updating from %s", self._request.url)
         try:
             with requests.Session() as sess:
                 response = sess.send(
-                    self._request, timeout=self._timeout, verify=self._verify_ssl
+                    self._request.prepare(), timeout=self._timeout, verify=self._verify_ssl
```

Comment:
In this case `prepare()` won't save us anything. I suggest that we just call `getattr(sess, method)(url…)`

Reply:
Sure. I was just trying to change as little code as possible - happy to refactor if you wish.

Reply:
@balloob Would you like me to preserve existing behaviour for non-digest authentication? Presumably this will perform a tiny bit better even if it isn't significant.